src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs

```diff
@@ -264,7 +264,17 @@ public override int Read(Span<byte> buffer)
             ThrowIfDisposed();
             if (!CanRead) throw new InvalidOperationException(SR.net_writeonlystream);
 
-            int bytesRead = _streamSocket.Receive(buffer, SocketFlags.None, out SocketError errorCode);
+            int bytesRead;
+            SocketError errorCode;
+            try
+            {
+                bytesRead = _streamSocket.Receive(buffer, SocketFlags.None, out errorCode);
+            }
+            catch (Exception exception) when (!(exception is OutOfMemoryException))
```

Comment:
If the goal is consistency with the array-based overload, why do the catch blocks differ?

Reply:
These overloads do not throw `SocketException`, but return `SocketError` as `out` parameter instead.

Reply:
We can refactor all overloads to use either the throwing or the `out` variant if we really care, but I would do it in a separate PR.

Reply:
I do think it would be better to be consistent here, and use the throwing versions consistently. That said, I'm fine not doing this in this PR.